### PR TITLE
GH-1656 Service enhancer: bug fix (order + limit), default union graph

### PR DESCRIPTION
**Pull request by Aklakan**
Fixed issues mentioned in #1656

 - [x] Tests are included.
 - ~~Documentation change and updates are provided for the [Apache Jena website](https://github.com/apache/jena-site/)~~ (no change in functionality)
 - [x] Commits have been squashed to remove intermediate development commit messages.
 - [x] Key commit messages start with the issue number (GH-xxxx or JENA-xxxx)

By submitting this pull request, I acknowledge that I am making a contribution to the Apache Software Foundation under the terms and conditions of the [Contributor's Agreement](https://www.apache.org/licenses/contributor-agreements.html).

### CODE REVIEW

File: jena-extras/jena-serviceenhancer/src/test/java/org/apache/jena/sparql/service/enhancer/impl/TestServiceEnhancerBatchQueryRewriter.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+public class TestServiceEnhancerBatchQueryRewriter {
+
+    /**
+     * Test that the generation of a bulk request from a query that uses limit and order by uses the correct rewrite.
+     * Specifically, ORDER BY must be retained on the union members even if they appear without slice because
+     * it might make a difference when the result set is cut off by the service's result set limit.
+     */
+    @Test
+    public void testOrderBy_01() {
```
Comment (Aklakan) [inline jena-extras/jena-serviceenhancer/src/test/java/org/apache/jena/sparql/service/enhancer/impl/TestServiceEnhancerBatchQueryRewriter.java:43]: This test case checks that offset / limit is correctly considered when converting bulk request (represented as a "batch") into a SPARQL union.

File: jena-extras/jena-serviceenhancer/src/test/java/org/apache/jena/sparql/service/enhancer/assembler/TestServiceEnhancerDatasetAssembler.java [hunk 2]
```diff
@@ -115,4 +124,54 @@ public void testAssemblerMgmtSuccess() {
...
         }
     }
+
+    /**
+     * Test case to check that when {@code OpService('loop:', subOp)} is evaluated against an union default
+     * graph (UDG) enabled dataset then the subOp must be evaluated on the UDG too.
+     * This also applies to {@code cache:} and {@code bulk:}.
+     */
+    @Test
+    public void testAssemblerTdbUnionDefaultGraph() throws IOException {
```
Comment (Aklakan) [inline jena-extras/jena-serviceenhancer/src/test/java/org/apache/jena/sparql/service/enhancer/assembler/TestServiceEnhancerDatasetAssembler.java:134]: This test case uses the service enhancer with an active dataset that is a default union graph. The pattern inside the service clause is expected to be evaluated against the default union graph as well.

**Comment by Aklakan** [comment]
This PR fixes the issues of #1656 and adds the corresponding test cases as per comments.

**Comment by afs** [comment]
First pass - this looks OK and I hope it will be in 4.7.0.

**Review by afs — APPROVED** [review]

**Pull request merged**